# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../fusionsound/ffmpeg-fix.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From: Ville Syrjala <syrjala@sci.fi>
Date: Tue, 21 Oct 2008 13:17:17 +0000 (+0300)
Subject: ffmpeg: Follow ffmpeg changes
X-Git-Url: http://git.directfb.org/?p=core%2FFusionSound.git;a=commitdiff_plain;h=6c9f3e777453b0cdbdf3e1db57ae4bfd7dd922d5

ffmpeg: Follow ffmpeg changes
---

diff --git a/configure.in b/configure.in
index a9ba0b1..3fab648 100644
--- a/configure.in
+++ b/configure.in
@@ -523,11 +523,11 @@ AC_ARG_WITH(ffmpeg,
     
 if test "x$with_ffmpeg" = "xyes"; then
   AC_MSG_CHECKING([for libavcodec/libavformat])
-  if $PKG_CONFIG libavcodec libavformat; then
+  if $PKG_CONFIG libavcodec libavformat libavutil; then
     AC_MSG_RESULT(yes)
     ffmpeg="yes"
-    FFMPEG_CFLAGS=`$PKG_CONFIG --cflags libavcodec libavformat`
-    FFMPEG_LIBS=`$PKG_CONFIG --libs libavcodec libavformat`
+    FFMPEG_CFLAGS=`$PKG_CONFIG --cflags libavcodec libavformat libavutil`
+    FFMPEG_LIBS=`$PKG_CONFIG --libs libavcodec libavformat libavutil`
   else
     AC_MSG_RESULT(no)
     AC_MSG_WARN([
diff --git a/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c b/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
index 2a99a16..6ec5317 100644
--- a/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
+++ b/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
@@ -41,8 +41,8 @@
 
 #include <misc/sound_util.h>
 
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 
 static DirectResult
 Probe( IFusionSoundMusicProvider_ProbeContext *ctx );
@@ -123,8 +123,8 @@ av_read_callback( void *opaque, uint8_t *buf, int size )
      return len;
 }
 
-static offset_t
-av_seek_callback( void *opaque, offset_t offset, int whence )
+static int64_t
+av_seek_callback( void *opaque, int64_t offset, int whence )
 {
      IFusionSoundMusicProvider_FFmpeg_data *data = opaque;
      unsigned int                           pos  = 0;
